piton/verif/env/manycore/manycore_top.v.pyv

```diff
@@ -570,6 +570,48 @@ system system(
             $display("UART: TX changed to %d at", tx, $time);
         end*/
 
+`ifdef ENABLE_COSIM
+    // Bind cosim module
+    bind bp_be_top
+      bp_nonsynth_cosim
+      #(.bp_params_p(bp_params_p))
+      cosim
+        (.clk_i(clk_i)
+           ,.reset_i(~cmp_top.sys_rst_n)
+           ,.freeze_i(calculator.pipe_sys.csr.cfg_bus_cast_i.freeze)
+
+           // We want to pass these values as parameters, but cannot in Verilator 4.025
+           // Parameter-resolved constants must not use dotted references
+           ,.cosim_en_i(1'b1)
+           ,.trace_en_i(1'b1)
+           ,.checkpoint_i(1'b0)
+           ,.num_core_i(1)
+           ,.mhartid_i(calculator.pipe_sys.csr.cfg_bus_cast_i.core_id)
+           ,.config_file_i("prog.cfg")
+           ,.instr_cap_i(0)
+           ,.memsize_i(256)
+           ,.amo_en_i(1)
+
+           ,.decode_i(calculator.reservation_n.decode)
+
+           ,.is_debug_mode_i(calculator.pipe_sys.csr.is_debug_mode)
+           ,.commit_pkt_i(calculator.commit_pkt_cast_o)
+
+           ,.priv_mode_i(calculator.pipe_sys.csr.priv_mode_r)
+           ,.mstatus_i(calculator.pipe_sys.csr.mstatus_lo)
+           ,.mcause_i(calculator.pipe_sys.csr.mcause_lo)
+           ,.scause_i(calculator.pipe_sys.csr.scause_lo)
+
+           ,.ird_w_v_i(scheduler.iwb_pkt_cast_i.ird_w_v)
+           ,.ird_addr_i(scheduler.iwb_pkt_cast_i.rd_addr)
+           ,.ird_data_i(scheduler.iwb_pkt_cast_i.rd_data)
+
+           ,.frd_w_v_i(scheduler.fwb_pkt_cast_i.frd_w_v)
+           ,.frd_addr_i(scheduler.fwb_pkt_cast_i.rd_addr)
+           ,.frd_data_i(scheduler.fwb_pkt_cast_i.rd_data)
+           );
+`endif // ENABLE_COSIM
+
```

Comment:
Could we make this macro a bit more BP-specific maybe?

Reply:
I'm just going to take this out, cosim never worked quite right